image_to_table/image_with_word_boxes.py

```diff
@@ -1,146 +1,87 @@
-    thickness = 2
-    if boxes is None:
-        boxes = []
-    for box in boxes:
-        image = cv2.rectangle(
-            image, box.lower_left_corner.as_tuple(), box.upper_right_corner.as_tuple(), color, thickness
-        )
-    cv2.imshow("table", image)
-    while True:
-        key = cv2.waitKey(0)
-        key_is_esc = key == 27
-        if key_is_esc:
-            break
-    cv2.destroyAllWindows()
-
-
-def boxes_in_x(boxes, x):
-    return [box for box in boxes if box.x_inside(x)]
-
-
-def boxes_in_y(boxes, y):
-    return [box for box in boxes if box.y_inside(y)]
-
-
-def make_row_boxes(y_counts, max_x):
-    inside = False
-    starts = []
-    stops = []
-
-    for y, n in enumerate(y_counts):
-        if n > 0 and not inside:
-            starts.append(y - 1)
-            inside = True
-        if n == 0 and inside:
-            stops.append(y)
-            inside = False
-    if inside:
-        stops.append(y)
-
-    box_ys = list(zip(starts, stops))
-    row_boxes = [
-        Box(
-            lower_right_corner=Point(x=max_x, y=stop),
-            lower_left_corner=Point(x=0, y=stop),
-            upper_right_corner=Point(x=max_x, y=start),
-            upper_left_corner=Point(x=0, y=start),
-        )
-        for (start, stop) in box_ys
-    ]
-    return row_boxes
-
-
-def make_column_boxes(x_counts, max_y):
-    inside = False
-    starts = []
-    stops = []
-
-    for x, n in enumerate(x_counts):
-        if n > 0 and not inside:
-            starts.append(x - 1)
-            inside = True
-        if n == 0 and inside:
-            stops.append(x)
-            inside = False
-    if inside:
-        stops.append(x)
-
-    box_xs = list(zip(starts, stops))
-    column_boxes = [
-        Box(
-            lower_right_corner=Point(y=max_y, x=stop),
-            lower_left_corner=Point(y=0, x=stop),
-            upper_right_corner=Point(y=max_y, x=start),
-            upper_left_corner=Point(y=0, x=start),
-        )
-        for (start, stop) in box_xs
-    ]
-    return column_boxes
-
-
-def show_all_boxes_intersecting(filename, row_boxes, column_boxes):
-    for row_box in row_boxes:
-        for column_box in column_boxes:
-            show_image_with_word_boxes(filename, [row_box, column_box])
-
-
-def get_row(row_box, column_boxes, original_boxes):
-    new_row = [get_cell_in_row(column_box, original_boxes, row_box) for column_box in column_boxes]
-    return new_row
-
-
-def get_cell_in_row(column_box, original_boxes, row_box):
-    row_column = []
-    for i, original_box in enumerate(original_boxes):
-        if original_box.is_inside_box(column_box) and original_box.is_inside_box(row_box):
-            row_column.append(original_box.text)
-    return " ".join(row_column)
-
-
-def extract_table_from_image(filename):
+
+def merge_sorted_text_boxes(boxes: List[TextBox]) -> TextBox:
+    bounding_rect = boxes[0].rect or boxes[-1].rect
+    description = " ".join(map(lambda x: x.text, boxes))
```

Comment:
`" ".join(x.text for x in boxes)` would work as well, and might be a bit more readable imo